### PR TITLE
83 Google Drive Upload

### LINKED ISSUE #83: Google Sheet Intergration
**Is your feature request related to a problem? Please describe.**
It would be nice to have spreadsheets on the RMS linked to a project.

**Describe the solution you'd like**
The ability to embed google documents into CMS pages as well as upload from google docs (a bit like slack) into project/location/asset files.

**Describe alternatives you've considered**
Using tables on the CMS. It's not as pretty.

### PR DESCRIPTION
**Pull request by Robert-Watts**
This is the PR for issue 83. I have worked on this for a long time but its finally getting there. This may not be as useful as I thought as in the current implementation it is limited to files of 10mb as this is what the google API call I am using supports ([see here for more info](https://developers.google.com/drive/api/v3/reference/files/export)), however this will cover most google docs. I do have some ideas to fix this but do not intend to for not therefore we need to come up with a way to communicate this to the user. I have not tested it yet with exports grater than 10mb. There are a few bits that need sorting before the is merged.

**If you are working on one of these tasks could you please comment on the PR so we know and don't work on the same thing concurrently.**

- [x] Create Basic Class
- [x] Sort mime types and blobs in the converter after a file is
- [ ] Update ReadMe to include the new endpoints and API
- [x] File name is missing when added to S3 Table
- [x] Save token in browser so that you doing need to keep logging in [See here for how to load an oauth token](https://www.gmass.co/blog/google-picker-api/)
- [x] Fix logout function - when you log back in it defaults to the same account giving you no choice
- [x] Loading Screen
- [ ] Make sure its class is only included by twig once per page
- [x] Test with files grater than 10mb
- [x] Create a test google folder with all reasonable files in and test the uploader against each one

### CODE REVIEW

File: html/admin/common/plugins/uppy.twig [hunk 1]
```diff
@@ -85,11 +88,16 @@ x-amz-meta-size
...
                 { id: 'credit', name: 'Credit', placeholder: 'How should the image be credited? Don\'t include "Image:"' },
-                { id: 'ownwork', name: 'Photo from a member of the Nouse Team?', render: function({value, onChange}, h) {
+                { id: 'ownwork', name: 'Photo from a member of the TechSoc Team?', render: function({value, onChange}, h) {
                         return h('input', { type: 'checkbox', onChange: (ev) => onChange(ev.target.checked ? 'on' : 'off'), defaultChecked: value === 'off' })
                     } }
                 {% endif %}
             ]
+        }).use(UppyGoogleDrive, {
+            target: Uppy.Dashboard,
+            google_client_id: "{{ CONFIG["AUTH-PROVIDERS"]["GOOGLE"]["keys"]['id'] }}",
```
Comment (Robert-Watts) [inline html/admin/common/plugins/uppy.twig:98]: @Jbithell Would you like to use the same google key as the login or a different one just for uppy?

**Comment by Robert-Watts** [comment]
No longer a need to have a  Loading Screen as using PreProcessor instead

**Comment by Robert-Watts** [comment]
Don't think its worth saving the token in the browser.

**Comment by Robert-Watts** [comment]
See the test files for this https://drive.google.com/drive/folders/1flvry2DEUtlQ0jNZuIq6QDKSHdYuEFtp?usp=sharing

**Comment by Jbithell** [comment]
How not to do a merge commit, by @Jbithell

**Comment by Jbithell** [comment]
I'm going to go ahead and close this for now as some housekeeping. Significant work would be required to merge main back into this PR.

**Comment by Jbithell** [comment]
I'm going to go ahead and close this for now as some housekeeping. Significant work would be required to merge main back into this PR.

**Pull request closed without merging**